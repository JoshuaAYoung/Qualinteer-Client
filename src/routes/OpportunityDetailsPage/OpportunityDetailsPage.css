.detailsHeader {
margin-right: 25rem;
margin-bottom: 2rem;
}

.detailsBanner {
  position: relative;
  height: 24rem;

}

.detailsInstructions {
  margin: 0 20rem 0 0;
}

.orgTitle {
  font-weight: 500
}

.detailsImage {
  width: 17rem;
  position: absolute;
  right: 1rem;
  top: 2rem;
}

.detailsContainer {
  background-color: #FFF0B0;
  padding: 2rem 2rem 1rem;
  width: auto;
  font-size: 1.4rem;
  margin-bottom: -6rem;
  padding-bottom: 5rem;
}

.detailsSection {
  position: relative;
  margin: 0 -2.5rem;
}

.detailsTitle {
  font-size: 2.2rem;
  margin: 0 0 1rem;
}

.opportunityInfo {
  margin-bottom: 1.3rem;
}

.volunteerLink {
  text-decoration: none;
  color: white;
}

.volunteerButton {
  width: auto;
  display: inline-block;
  height: 3.6rem;
  cursor: pointer;
  background-color: #F0667B;
  font-family: 'Montserrat', sans-serif;
  font-weight: 500;
  font-size: 1.7rem;
  cursor: pointer;
  border-radius: 1.4rem;
  padding: .7rem 2rem 0;
  transition: all 0.3s;
  margin-bottom: 2rem;
}

.volunteerButton:hover {
  color: black;
  background-color: #F4BD5B;
}

.opportunityDescription {
  margin-top: 1rem;
}


@media only screen and (max-width: 360px) {
  .detailsInstructions {
    margin: 0 16rem 0 0;
  }

  .detailsImage {
    right: 0
  }
}

@media only screen and (min-width: 420px) {
  .detailsBanner {
    margin-right: 10%;
    margin-left: 10%;
  }
}



@media only screen and (min-width: 575px) {
  .detailsContainer {
    border-top: .2rem solid #F4BD5B;
    border-bottom: .2rem solid #F4BD5B;
    margin-bottom: 7rem;
    padding-bottom: 0;
    margin-left: 10%;
    margin-right: 10%;
  }
  
  .detailsImage {
    width: 22rem;
  }
  
  .detailsBanner {
    margin-right: 15%;
    margin-left: 15%;
    margin-bottom: 5rem;
  }

  .detailsInstructions {
    margin-right: 28rem;
  }

  .detailsHeader {
    margin-bottom: 2rem;
    line-height: 6rem;
    font-size: 4.8rem;
  }
}

@media only screen and (min-width: 768px) {
  .detailsBanner {
    margin-right: 20%;
    margin-left: 20%;
    margin-bottom: 5rem;
  }
  
  .detailsInstructions {
    margin-right: 26rem;
  }

  .detailsHeader {
    margin-bottom: 4rem;
  }
}

@media only screen and (min-width: 1024px) {
  .detailsInstructions {
    margin-right: 26rem;
  }

  .detailsContainer {
    margin-left: 0;
    margin-right: 0;
  }
  
  .descriptionContainer {
    margin-left: 0;
    margin-right: 0;
  }
}

