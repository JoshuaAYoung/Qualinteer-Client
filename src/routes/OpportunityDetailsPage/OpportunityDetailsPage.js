import React, { Component } from 'react';
import OpportunityDetails from '../../components/OpportunityDetails/OpportunityDetails'
import OppApiService from '../../services/opp-api-service'
import VableContext from '../../context/VableContext'
import './OpportunityDetailsPage.css'
import ErrorToast from '../../components/ErrorToast/ErrorToast'
import TokenService from '../../services/token-service'
import ScrollToTopOnMount from '../../utilities/ScrollToTopOnMount'

export default class OpportunityBoardPage extends Component {
  static contextType = VableContext;

  state = { error: null }

  renderInstructions = () => {
    if (this.context.userType === 'volunteer' && TokenService.hasAuthToken()) {
      return (<p className='pageInstructions detailsInstructions'>Click on the "Volunteer" button to apply.</p>)
    }
    else {
      return (<p className='pageInstructions detailsInstructions'>Login as a volunteer user to apply.</p>)
    }
  }

  componentDidMount() {
    const { oppId } = this.props.match.params
    this.setState({ error: null })
    OppApiService.getOppById(oppId)
      .then(this.context.setOpportunity)
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <div className='opportunityDetails'>
        <ScrollToTopOnMount />
        <div className="detailsBanner">
          <h1 className='pageHeader detailsHeader'>Opportunity Details.</h1>
          <img src='/assets/board-rocket.svg' alt='people working on a spaceship' className='detailsImage' />
          {this.renderInstructions()}
        </div>
        {error && <ErrorToast errorMessage={error} />}
        <OpportunityDetails history={this.props.history} />
      </div>
    );
  }
}
