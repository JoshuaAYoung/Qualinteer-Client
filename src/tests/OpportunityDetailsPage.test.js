import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import OpportunityDetailsPage from '../routes/OpportunityDetailsPage/OpportunityDetailsPage';

describe('OpportunityDetailsPage Component', () => {
  const props = {
    match: {
      params: '1'
    }
  }

  it('OpportunityDetailsPage Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<OpportunityDetailsPage {...props} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('OpportunityDetailsPage Snapshot Test', () => {
    const wrapper = shallow(<OpportunityDetailsPage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})