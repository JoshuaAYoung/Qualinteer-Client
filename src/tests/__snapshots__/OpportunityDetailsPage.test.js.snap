// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityDetailsPage Component OpportunityDetailsPage Snapshot Test 1`] = `
<div
  className="opportunityDetails"
>
  <ScrollToTopOnMount />
  <div
    className="detailsBanner"
  >
    <h1
      className="pageHeader detailsHeader"
    >
      Opportunity Details.
    </h1>
    <img
      alt="people working on a spaceship"
      className="detailsImage"
      src="/assets/board-rocket.svg"
    />
    <p
      className="pageInstructions detailsInstructions"
    >
      Login as a volunteer user to apply.
    </p>
  </div>
  <OpportunityDetails />
</div>
`;
